### PR TITLE
Anya/6953 create remand reasons

### LINKED ISSUE #6736: Add issue dispositions for AMA appeals
When attorneys are assigning AMA cases to judges, they need to add issue dispositions for each issue, so the judge can review the draft decision.

**Out of scope:** 
- Attorneys editing issues
- Attorneys adding dispositions to edited issues

### LINKED ISSUE #6953: [BE] Implement a way for attorneys and judges to update remand reasons for AMA appeals

### LINKED ISSUE #7044: Remand reason selections disappear after changing the disposition.
If disposition is pre-selected remanded, you see that remand reasons are pre-selected. If you change the disposition to allowed and then back to remanded, remand reasons selections disappear. 

Steps to reproduce:
1. Attorney complete draft decision and set remanded disposition for an issue with remand reasons.
2. Log in as judge, check that disposition is remanded and remand reasons are selected
3. Change disposition to allowed and then back to remanded, go to the remand reasons page and not remand reasons are not pre-selected.

### PR DESCRIPTION
**Pull request by aroltsch**
Resolves #6953 
Resolves #7044
Resolves #6736

Allow a judge or an attorney to update issue dispositions and remand reasons.

Test Cases:
1. If disposition is remanded and has remand reasons and an attorney/judge changes the disposition, remand reasons must be deleted on the BE
2. If disposition is changed to remanded, remand reasons must be selected
3. If disposition is remanded, an attorney/judge should be presented with a way to change remand reasons:
- if remand reason A is in the DB but is not selected by the user, remand reason A must be deleted
- if remand reason B is in the DB and is selected by the user, post AOJ boolean must be updated
- if remand reason C is not in the DB and is selected by the user, remand reason C must be created
4. If disposition is remanded and has remand reasons set, changing disposition to `!remanded` and back to `remanded` (while staying in checkout flow) does not delete remand reasons (#7044) (i.e. remand reasons are not cleared from `state.queue.stagedChanges` until the user finishes the checkout flow)

### CODE REVIEW

File: app/controllers/issues_controller.rb [hunk 1]
```diff
@@ -42,7 +42,7 @@ def json_issues
     appeal.issues.map do |issue|
       ActiveModelSerializers::SerializableResource.new(
         issue,
-        serializer: ::WorkQueue::IssueSerializer
+        serializer: ::WorkQueue::LegacyIssueSerializer
```
Comment (aroltsch) [inline app/controllers/issues_controller.rb]: Renamed this serializer to be consistent with the rest of the legacy classes

File: app/models/concerns/case_review_concern.rb [hunk 2]
```diff
@@ -21,15 +21,35 @@ def update_task_and_issue_dispositions
     if task.type == "AttorneyTask" && task.assigned_by_id != reviewing_judge_id
       task.parent.update(assigned_to_id: reviewing_judge_id)
     end
-    (issues || []).each do |issue|
-      # TODO: update request issues for RAMP appeals for now. When we build out
-      # decision issues further, we'll update those.
-      # decision_issue = appeal.decision_issues.find_by(id: issue["id"]) if appeal
-      request_issue = appeal.request_issues.find_by(id: issue["id"]) if appeal
-
-      # decision_issue.update(disposition: issue["disposition"]) if decision_issue
```
Comment (aroltsch) [inline app/models/concerns/case_review_concern.rb]: We are leaving these comments on purpose.

File: app/models/concerns/case_review_concern.rb [hunk 3]
```diff
@@ -21,15 +21,35 @@ def update_task_and_issue_dispositions
...
+
+  def update_remand_reasons(request_issue, remand_reasons_attrs)
+    remand_reasons_attrs.each do |remand_reason_attrs|
+      request_issue.remand_reasons.find_or_initialize_by(code: remand_reason_attrs["code"]).tap do |record|
+        record.post_aoj = remand_reason_attrs["post_aoj"]
+        record.save!
+      end
     end
+    # Delete remand reasons that are not part of the request
+    existing_codes = request_issue.reload.remand_reasons.pluck(:code)
```
Comment (mdbenjam) [inline app/models/concerns/case_review_concern.rb]: An alternative here that might simplify this is to delete the remand reasons first, then add the ones in the request, all in a transaction.

Comment (mdbenjam) [reply app/models/concerns/case_review_concern.rb]: Also this ticket confuses me: https://github.com/department-of-veterans-affairs/caseflow/issues/7044 Are we trying to delete the old remand reasons, or not?

File: client/app/queue/components/SelectIssueDispositionDropdown.jsx [hunk 4]
```diff
@@ -56,18 +74,13 @@ class SelectIssueDispositionDropdown extends React.PureComponent<Props> {
...
-        options={Object.entries(VACOLS_DISPOSITIONS_BY_ID).slice(0, 6).
-          map((opt) => ({
-            label: `${opt[0]} - ${String(opt[1])}`,
-            value: opt[0]
-          }))}
+        options={this.getDispositions()}
         onChange={(option) => this.props.updateIssue({
           disposition: option ? option.value : null,
-          readjudication: false,
-          remand_reasons: []
```
Comment (mdbenjam) [inline client/app/queue/components/SelectIssueDispositionDropdown.jsx]: So this is to make it if you switch back to remand, that the reasons stay selected?

Comment (dannysellers) [reply client/app/queue/components/SelectIssueDispositionDropdown.jsx]: Correct. Without this, if you set an issue disposition to Remanded, set remand reasons, go back to dispositions page, and change that issue's disposition, we'd wipe all its remand reasons. Now, we save the reasons in Redux until the very end (until `utils.buildCaseReviewPayload`), only sending remand reasons in the payload for remanded issues

**Pull request merged**